front/app/containers/Authentication/useSteps/stepConfig/missingDataFlow.ts

```diff
@@ -135,6 +155,45 @@ export const missingDataFlow = (
             onSuccess: async () => {
               const { requirements } = await getRequirements();
 
+              if (showOnboarding(requirements.onboarding)) {
+                setCurrentStep('missing-data:onboarding');
+                return;
+              }
+
+              if (requirements.special.group_membership === 'require') {
+                setCurrentStep('closed');
+                return;
+              }
+
+              setCurrentStep('success');
+            },
+          }
+        );
+      },
+      SKIP: async () => {
+        const { requirements } = await getRequirements();
+
+        if (showOnboarding(requirements.onboarding)) {
+          setCurrentStep('missing-data:onboarding');
+          return;
+        }
+
+        setCurrentStep('success');
+      },
+    },
+
+    'missing-data:onboarding': {
+      CLOSE: () => setCurrentStep('closed'),
+      SUBMIT: async (
+        userId: string,
+        onboarding: Record<string, RequirementStatus>
+      ) => {
+        updateUser(
```

Comment:
The way `updateUser` is used everywhere in the step configuration is incorrect- it should be an `async` function in order for the loading states and transitions to work correctly. This seems to be a regression from during the `react-query` conversion. Took the liberty to fix it myself here: https://github.com/CitizenLabDotCo/citizenlab/pull/5712/files